data_pipelines/resources/io_managers.py

```diff
@@ -16,46 +17,56 @@
 from fsspec.implementations.local import LocalFileSystem
 from upath import UPath
 
+from data_pipelines.settings import settings
 from data_pipelines.utils.flood.config import USE_CONTROL_MEMBER_IN_ENSEMBLE
 
-from .rio_session import RIOSession
 
+def get_path_in_asset(
+    context: AssetExecutionContext, base_path: UPath, extension: str
+) -> UPath:
+    path = base_path.joinpath(*context.asset_key.path)
+    if context.has_partition_key:
+        if len(context.partition_keys) == 1:
+            path = path / context.partition_key
+        else:
+            raise NotImplementedError(
+                "get_path_in_asset does not support multi-partition runs."
+            )
+    return path.with_suffix(extension)
 
-class COGIOManager(ConfigurableIOManager):
-    base_path: str
-    rio_env: ResourceDependency[RIOSession]
 
-    def get_path(self, context: InputContext | OutputContext) -> str:
-        return os.path.join(
-            self.base_path,
-            *context.asset_key.path,
-            f"{context.partition_key}.tif",
-        )
+class COGIOManager(UPathIOManager):
+    extension: str = ".tif"
 
-    def handle_output(
-        self, context: OutputContext, data: xr.DataArray | xr.Dataset | None
-    ) -> None:
-        context.log.debug("COGIOManager was used.")
-        if data is None:
-            context.log.info(
-                "Received value of None for the data argument in COGIOManager.handle_output. "
-                "Skipping output handling, assuming that the output was handled within the asset definition."
-            )
-            return
+    def __init__(self, base_path: UPath):
+        super().__init__(base_path=base_path)
 
-        path = self.get_path(context)
-        data.to_raster(path)
+    def get_path_in_asset(self, context: AssetExecutionContext) -> UPath:
+        path = self._base_path.joinpath(context.asset_key.path)
+        if context.has_partition_key:
+            if len(context.partition_key) == 1:
+                path = path / context.partition_key
+            else:
+                raise NotImplementedError(
+                    "get_path_in_asset does not support multi-partition runs."
+                )
+        return path.with_suffix(self.extension)
```

Comment:
Yes good point. I actually meant to remove the method inside the class but this might be better, as the method will always use the same base_path and extension as the assets io_manager.